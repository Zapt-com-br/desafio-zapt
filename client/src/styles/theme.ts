export default {
  grid: {
    container: '124.8rem',
    gutter: '2.0rem'
  },
  border: {
    radius: '0.4rem'
  },
  font: {
    family:
      "'Inter', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",
    light: 300,
    normal: 400,
    medium: 500,
    sizes: {
      xsmall: '1.2rem',
      small: '1.4rem',
      medium: '1.6rem',
      normal: '1.8rem',
      large: '2.0rem',
      xlarge: '2.8rem',
      huge: '4.0rem'
    }
  },
  colors: {
    primary: '#334BC8',
    secondary: '#FF5B58',
    white: '#FFFFFF',
    black: '#000000',
    drWhite: '#FAFAFA',
    namaraGrey: '#7B7B7B'
  },
  spacings: {
    xxsmall: '0.8rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '5.6rem'
  },
  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50
  }
} as const
