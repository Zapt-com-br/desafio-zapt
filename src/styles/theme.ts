export default {
  colors: {
    primary: '#011FBB',
    primaryLight: '#334BC8',
    black: '#000000',
    white: '#ffffff',
    background: '#E5E5E5',
    bacgrounCard: '#FAFAFA',
    darkgray: '#7B7B7B',

    gradients: {
      lightBg:
        'linear-gradient(0deg, rgba(255, 255, 255, 0.9), rgba(255, 255, 255, 0.9)), #2196F3;',
    },
  },

  border: {
    radius: '0.4rem',
  },

  font: {
    family: 'Inter',
    weight: {
      '300': 300,
      '400': 400,
      '500': 500,
    },

    sizes: {
      xsmall: '1.2rem',
      small: '1.4rem',
      medium: '1.6rem',
      xmedium: '2rem',
      large: '2.8rem',
      xlarge: '4rem',
    },
  },

  spacings: {
    xxsmall: '0.8rem',
    xsmall: '1.6rem',
    small: '2.4rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
  },
} as const;
