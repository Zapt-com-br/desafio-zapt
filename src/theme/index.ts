import { keyframes } from 'styled-components'

export const T = {
  grid: {
    container: '130rem',
    gutter: '2.4rem'
  },
  border: {
    radius: '0.4rem',
    circle: '50%'
  },
  font: {
    family:
      "'Poppins', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",
    normal: 400,
    bold: 500,
    sizes: {
      xSmall: '1.2rem',
      small: '1.4rem',
      medium: '1.6rem',
      Large: '2.0rem',
      xLarge: '2.8rem',
      xxLarge: '4rem'
    }
  },
  colors: {
    primary: '#334BC8',
    secondary: '#fff',
    primaryHover: '#5872ff',
    gray: '#7A7A7A',
    grayHover: '#aaaaaa',
    grayLight: '#FAFAFA',
    white: '#fff',
    black: '#000',
    blue: '#011FBB',
    blueLight: '#E9F4FE',
    disabled: '#a1a4b0'
  },
  spacings: {
    xxSmall: '0.4rem',
    xSmall: '0.8rem',
    small: '1.2rem',
    medium: '1.6rem',
    large: '2rem',
    xLarge: '2.4rem',
    xxLarge: '2.8rem',
    xxxLarge: '3.2rem'
  },
  layers: {
    base: 10
  },
  transitions: {
    default: '0.3s ease-in-out'
  },
  animations: {
    rotate: keyframes`
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  `
  },
  breakPoint: (num1: number) => `@media screen and (min-width: ${num1}rem)`
} as const
