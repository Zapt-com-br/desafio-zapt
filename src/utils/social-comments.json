{
  "title": "O que est√£o falando nas redes sociais",
  "comments": [
    {
      "id": 1,
      "comment": "N√£o da para parar de comer.. esse final de semana pedi 20 pasteis da Ilma e acabaram TODOS em um s√≥ dia... O pessoal aqui de casa ficou viciado de um n√≠vel que terei que comprar toda semana... Assim n√£o tem academia que secura rsrsrsrs Obrigado Zapt.",
      "name": "ferri-gui",
      "image": "/images/social-comments/ferri-gui.svg"
    },
    {
      "id": 2,
      "comment": "Estava precisando achar algum lugar que vendia produtos que v√™em direto da fazenda... a Zapt salvou.. e ainda paquei um pre√ßo muito mais barato que supermecado üôå",
      "name": "paolaC",
      "image": "/images/social-comments/paola.svg"
    },
    {
      "id": 3,
      "comment": "Todos os vinhos da vin√≠cula Bueno est√£o dispon√≠veis no catalogo da Zapt a um pre√ßo incriv√©l... n√£o percam essa oportunidade...",
      "name": "Galv√£o",
      "image": "/images/social-comments/galvao.svg"
    },
    {
      "id": 4,
      "comment": "Nunca tinha achado os produtos na Lush depois que ela saiu do Brasil.. s√≥ a Zapt para conseguir importalos a pre√ßo mais barato do que vendiam aqui! Muito obrigado, recomendo sempre a todos os amigos.",
      "name": "LiciaR",
      "image": "/images/social-comments/liciar.svg"
    },
    {
      "id": 5,
      "comment": "Ontem chegou uma pizza que comprei na Zapt e me lembrou as que minha av√≥ fazia... sensacional!",
      "name": "shitz",
      "image": "/images/social-comments/shitz.svg"
    },
    {
      "id": 6,
      "comment": "Nossa cada dia tem uma oferta mais legal do que a outra.. Al√©m de produtos de qualidade o pr√©co √© imbat√≠vel",
      "name": "rafael",
      "image": "/images/social-comments/rafael.svg"
    },
    {
      "id": 7,
      "comment": "Achei uma promo√ß√£o incr√≠vel de roupinha de beb√™, foi o que salvou o enxoval do Pedro.",
      "name": "gust",
      "image": "/images/social-comments/gust.svg"
    },
    {
      "id": 8,
      "comment": "Meu dono finalmente conseguiu achar uma comida que eu gosto! Agora vou conseguir engordar e ser feliz.",
      "name": "Cookie",
      "image": "/images/social-comments/cookie.svg"
    },
    {
      "id": 9,
      "comment": "N√£o conhecia a Zapt.. um amigo meu indicou... falou para eu dar uma olhada nos pre√ßos das cervejas artesanais.. incr√≠vel.",
      "name": "duque",
      "image": "/images/social-comments/duque.svg"
    }
  ]
}
