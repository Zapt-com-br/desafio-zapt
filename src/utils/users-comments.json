{
  "title": "O que as pessoas estão achando",
  "comments": [
    {
      "id": 1,
      "image": "/images/user-comments/image-1.png",
      "comment": "Não consigo mais viver sem os produtos do Sítio, a diferença do que eu comprava antes e agora é abissal",
      "user": {
        "avatar": "/images/user-comments/patricia.svg",
        "name": "Patricia Godoy",
        "city": "São Paulo",
        "state": "São Paulo"
      }
    },
    {
      "id": 2,
      "image": "/images/user-comments/image-2.png",
      "comment": "Depois que você começa a usar produtos de qualidade é bem difícil voltar atrás...",
      "user": {
        "avatar": "/images/user-comments/marcinho.svg",
        "name": "Marcinho Ribeiro",
        "city": "Belo Horizonte",
        "state": "Minas Gerais"
      }
    },
    {
      "id": 3,
      "image": "/images/user-comments/image-1.png",
      "comment": "Não consigo mais viver sem os produtos do Sítio, a diferença do que eu comprava antes e agora é abissal",
      "user": {
        "avatar": "/images/user-comments/patricia.svg",
        "name": "Patricia Godoy",
        "city": "São Paulo",
        "state": "São Paulo"
      }
    },
    {
      "id": 4,
      "image": "/images/user-comments/image-2.png",
      "comment": "Depois que você começa a usar produtos de qualidade é bem difícil voltar atrás...",
      "user": {
        "avatar": "/images/user-comments/marcinho.svg",
        "name": "Marcinho Ribeiro",
        "city": "Belo Horizonte",
        "state": "Minas Gerais"
      }
    }
  ]
}
