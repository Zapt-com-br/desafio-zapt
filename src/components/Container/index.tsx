import * as S from './styles';

export type ContainerProps = {
  children: React.ReactNode;
  as?: React.ElementType;
};

const Container = ({ children, as }: ContainerProps) => (
  <S.Wrapper as={as}>
    <S.Container>{children}</S.Container>
  </S.Wrapper>
);

export default Container;
