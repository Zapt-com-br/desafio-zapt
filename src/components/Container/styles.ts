import styled from 'styled-components';
import * as ButtonStyles from 'components/Button/styles';

export const Wrapper = styled.section`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  position: relative;
`;

export const ButtonWrapper = styled.div`
  position: absolute;
  bottom: 4rem;

  ${ButtonStyles.Wrapper} + ${ButtonStyles.Wrapper} {
    margin-left: 0.8rem;
  }
`;
