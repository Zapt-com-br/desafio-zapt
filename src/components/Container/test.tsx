import { render, screen } from 'utils/test-utils';

import theme from 'styles/theme';

import Container from '.';

describe('<Container />', () => {
  it('should render the heading', () => {
    const { container } = render(
      <Container>
        <div data-testid="mock children">Desafio Zapt</div>
      </Container>
    );

    expect(screen.getByTestId('mock children').parentElement).toHaveStyleRule(
      'max-width',
      theme.grid.container
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
