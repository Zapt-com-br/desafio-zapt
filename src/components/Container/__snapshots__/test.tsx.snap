// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should render the heading 1`] = `
.c0 {
  width: 100%;
}

.c1 {
  width: 90%;
  max-width: 120rem;
  margin-left: auto;
  margin-right: auto;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      data-testid="mock children"
    >
      Desafio Zapt
    </div>
  </div>
</div>
`;
